cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -329,9 +339,80 @@ public void apply() throws Exception {
   }
 
   /**
+   * Upgrades trigger and their state stored in the Store. It reads them as {@link TriggerStatus} through
+   * deserialization and then stores them back as {@link TriggerStatusV2} through custom serialization.
+   */
+  public void upgrade() throws InterruptedException, TransactionFailureException, IOException,
+    DatasetManagementException {
+    initializeScheduleTable();
+    factory.createExecutor(ImmutableList.of((TransactionAware) table))
+      .execute(new TransactionExecutor.Subroutine() {
+        @Override
+        public void apply() throws Exception {
+          Row result = table.get(TRIGGER_KEY);
+          if (!result.isEmpty()) {
+            boolean upgadedSchedules = false;
+            for (byte[] bytes : result.getColumns().values()) {
+              Object triggerObject = SerializationUtils.deserialize(bytes);
+              if (triggerObject instanceof TriggerStatus) {
+                TriggerStatus trigger = (TriggerStatus) triggerObject;
+                persistTrigger(table, trigger.trigger, trigger.state);
+                upgadedSchedules = true;
+                LOG.debug("Schedule: trigger with key {} upgraded", trigger.trigger.getKey());
+              }
+            }
+            if (upgadedSchedules) {
+              LOG.info("Schedule: Successfully upgraded required schedules.");
+            } else {
+              LOG.info("Schedule: No schedules needs to be upgraded.");
+            }
+          } else {
+            LOG.debug("Schedule: No schedules found in the store.");
```

Comment:
INFO. Also I think its ok to say "No schedules found in the store." without prepending with "Schedule: " everytime.